javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -345,6 +345,18 @@ private int filter(Predicate<? super T> predicate, Object results, int index, T
         return index;
     }
 
+    Object[] toArray() {
+        final Object[] results = new Object[length];
+        visit((index, leaf, start, end) -> {
+            final int copied = end - start;
+            for (int i = start; i < end; i++) {
+                results[index + i] = type.getAt(leaf, i);
```

Comment:
I want to avoid that we override conversion methods if possible.
All types could benefit if we change Value.toArray() instead.

```java
interface Value<T> extends Iterable<T> {
    
    default Object[] toJavaArray() {
        final Object[] results = new Object[length];
        int index = 0;
        for (T value : this) {
            results[index++] = value;
        }
        return results;
    }

}
```

Does it perform well?